### DISCUSSION TITLE
Desoto's calc params, near zero irradiance, and Rsh

### DISCUSSION THREAD

**Discussion by natecostello** [pvlib/pvlib-python#1978] [Q&A]
I've been messing around with pvlib for a couple weeks mainly looking at low irradiance behavior.  One thing that caught my eye was the discontinuity in Voc with the introduction of any amount of irradiance (vs zero).  This doesn't match some very crude empirical measurements I've made, so I started to focus on pvlib's behavior under those conditions.

What I've found as the primary responsible cause is the treatment/calculation of $R_{sh}$ by `pvlib.pvsystem.calcparams_desoto`

In that function he calculates $R_{sh} = \frac{E_{ref}}{E} \cdot R_{s-ref}$

At very low irradiance (like moonlight), this leads to a healthy $V_{oc}$ that doesn't match reality.

When I read desoto's [thesis](https://minds.wisconsin.edu/bitstream/handle/1793/7602/desoto04.pdf) he proposes not varying $R_{sh}$ based on his research.  Unfortunately, the paper referenced in the code is paywalled, so I can't see the basis for the change from the method in his thesis, assuming it is in there.

This also led me to take a close look at `pvlib.pvsystem.calcparams_pvsyst` which includes a parameter for shunt resistance at zero irradiance.  It also requires some parameters that don't seem to be offered by most manufacturers: gamma_ref and mu_gamma.  By estimating those two parameters and passing $R_{sh-ref}$ for shunt resistance at zero irradiance, the low irradiance curve matches the produced by `calc_params_desoto` modified by holding $R_{sh}$ constant.

**My question: Is there any good non-paywalled sources that I can be pointed towards that cover treatment of $R_{sh}$ at conditions of near zero irradiance**

Relevant Plot (a "12V" panel):
![image](https://github.com/pvlib/pvlib-python/assets/6625614/09ecd327-7a6f-41e6-8c90-4c0570bd5d3f)

**Comment by cwhanse** [comment]
@natecostello The pvlib code faithfully implements the algorithm in that Solar Energy 2006 paper. In that paper, the author points to analysis of NIST data (described but given [here](https://www.nist.gov/publications/evaluating-building-integrated-photovoltaic-performance-models)) as indicating an inverse relationship between effective irradiance and the slope of the IV curve near Isc. The author also points to a book, D. K. Schroder, “Semiconductor Material and Device Characterizations,” Willey-Interscience Publication, Arizona, 1998, as reaching the same conclusion.

At very low effective irradiance (e.g., moonlight on the order of 30 mW/m2), the single diode equation would predict that Voc is proportional to Rsh.

I doubt De Soto's data (or NIST data) considered irradiance much below ~200 W/m2, which is a reasonable minimum for daylight conditions.

**Comment by mikofski** [comment]
> When I read desoto's [thesis](https://minds.wisconsin.edu/bitstream/handle/1793/7602/desoto04.pdf) _he_ proposes...

FYI: Widalys De Soto-Burt is a woman.

**Reply by natecostello** [reply]
Thanks for the correction.  I'll leave that unedited as an example of implicit bias.
